### ISSUE TITLE
Random failure "DataProviderException: downloadEndpoint failed" in test_ocean_tx__create_url_asset::test1

### ISSUE THREAD

**Issue by trentmc** [oceanprotocol/ocean.py#1013]
NOTE: the failing test is [turned off](https://github.com/oceanprotocol/ocean.py/commit/c47d0e3841d071a64b7755203e704ec703695a0e), until we fix this issue. 

----

**Describe the bug**

In `tests/integration/ganache/test_ocean_tx__create_url_asset::test1`: sometimes (but not always), there's a failure: `ocean_lib.exceptions.DataProviderException: downloadEndpoint failed`

Logs: https://github.com/oceanprotocol/ocean.py/actions/runs/3251973453/jobs/5337655840

For convenience, here's the log of (a) test that fails, and (b) another test with a warning "out of MATIC", we can ignore it. Search for "downloadEndpoint" to find the failure.

```text
____________________________________ test1 _____________________________________

web3 = <web3.main.Web3 object at 0x7f2a8f82af10>
config = {'ADDRESS_FILE': '~/.ocean/ocean-contracts/artifacts/address.json', 'BLOCK_CONFIRMATIONS': 0, 'CHAIN_ID': 8996, 'DOWNLOADS_PATH': 'consume-downloads', ...}
publisher_wallet = <ocean_lib.web3_internal.wallet.Wallet object at 0x7f2a8fbc3e50>
consumer_wallet = <ocean_lib.web3_internal.wallet.Wallet object at 0x7f2a8fb30190>
tmp_path = PosixPath('/tmp/pytest-of-runner/pytest-0/test10')

    @pytest.mark.integration
    def test1(
        web3: Web3,
        config: dict,
        publisher_wallet: Wallet,
        consumer_wallet: Wallet,
        tmp_path,
    ):
        data_provider = DataServiceProvider
        ocean = Ocean(config)
    
        # Publish
        url = "https://cexa.oceanprotocol.io/ohlc?exchange=binance&pair=ETH/USDT"
        name = "CEXA ETH-USDT"
        (data_nft, datatoken, asset) = ocean.assets.create_url_asset(
            name, url, publisher_wallet
        )
    
        # Initialize service
        service = get_first_service_by_type(asset, ServiceTypes.ASSET_ACCESS)
        response = data_provider.initialize(
            did=asset.did, service=service, consumer_address=consumer_wallet.address
        )
    
        # Share access
        to_address = consumer_wallet.address
        datatoken.mint(to_address, ocean.to_wei(10), publisher_wallet)
    
        # Consume
        order_tx_id = ocean.assets.pay_for_access_service(asset, consumer_wallet)
>       file_path = ocean.assets.download_asset(
            asset=asset,
            consumer_wallet=consumer_wallet,
            destination=str(tmp_path),
            order_tx_id=order_tx_id,
        )

tests/integration/ganache/test_issue981_consume.py:51: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/site-packages/enforce_typing/decorator.py:29: in wrapper
    return func(*args, **kwargs)
ocean_lib/ocean/ocean_assets.py:804: in download_asset
    return download_asset_files(
/opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/site-packages/enforce_typing/decorator.py:29: in wrapper
    return func(*args, **kwargs)
ocean_lib/assets/asset_downloader.py:73: in download_asset_files
    data_provider.download(
/opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/site-packages/enforce_typing/decorator.py:29: in wrapper
    return func(*args, **kwargs)
ocean_lib/data_provider/data_service_provider.py:185: in download
    DataServiceProviderBase.check_response(
/opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/site-packages/enforce_typing/decorator.py:29: in wrapper
    return func(*args, **kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

response = <Response [400]>, endpoint_name = 'downloadEndpoint'
endpoint = 'http://172.15.0.4:8030/api/services/download'
payload = {'consumerAddress': '0x529043886F21D9bc1AE0feDb751e34265a246e47', 'documentId': 'did:op:6a7c302f3de107d067b4282c106bd771baeb2710e32334d851258b37892dc144', 'fileIndex': 0, 'nonce': '1665772103.76282', ...}
success_codes = None
exception_type = <class 'ocean_lib.exceptions.DataProviderException'>

    @staticmethod
    @enforce_types
    def check_response(
        response: Any,
        endpoint_name: str,
        endpoint: str,
        payload: Union[Dict, bytes],
        success_codes: Optional[List] = None,
        exception_type=DataProviderException,
    ):
        if not response or not hasattr(response, "status_code"):
            if isinstance(response, Response) and response.status_code == 400:
                error = response.json().get("error", None)
                if error is None:
                    error = response.json().get("errors", "unknown error")
    
>               raise DataProviderException(f"{endpoint_name} failed: {error}")
E               ocean_lib.exceptions.DataProviderException: downloadEndpoint failed: {}

ocean_lib/data_provider/base.py:303: DataProviderException
---------------------------- Captured stdout setup -----------------------------
sender: 0xc594c6e5def4bab63ac29eed19a134c130388f74f019bc74b8f4389df2837a58, 0xe2DD09d719Da89e5a3D0F2549c7E24566e947260, address: 0xe2DD09d719Da89e5a3D0F2549c7E24566e947260
private key: 0xc594c6e5def4bab63ac29eed19a134c130388f74f019bc74b8f4389df2837a58
public key: 0x8b1b222564fb8302e9acfa4bcd60060de03ae479fcbed38fe21e45cd39ddb3fefb60ba1b4625e00d02223027c8f485f6d77693ab62ce06289b23942c6b319ccf

sender balance: 999996.99999999994445071
----------------------------- Captured stdout call -----------------------------
signing message with nonce 1665772103.76282: did:op:6a7c302f3de107d067b4282c106bd771baeb2710e32334d851258b37892dc144, account=0x529043886F21D9bc1AE0feDb751e34265a246e47
----------------------------- Captured stderr call -----------------------------
2022-10-14 18:27:55 fv-az196-931 ocean[9022] INFO Successfully created NFT with address 0x99a64469d4bBaD1d79B3eb9BF131fAA8Ae25155a.
2022-10-14 18:27:55 fv-az196-931 ocean[9022] INFO Successfully created datatoken with address 0x9d9c6d87348c8345EF92AB77C9d30E7fB2A26c50.
2022-10-14 18:27:56 fv-az196-931 ocean_lib.data_provider.data_encryptor[9022] INFO Asset urls encrypted successfully, encrypted urls str: 0x04eca0bd...cff, encryptedEndpoint http://172.15.0.4:8030/api/services/encrypt
2022-10-14 18:27:56 fv-az196-931 ocean_lib.data_provider.data_encryptor[9022] INFO Asset urls encrypted successfully, encrypted urls str: 0x04cc035e...78074f7615, encryptedEndpoint http://172.15.0.4:8030/api/services/encrypt
2022-10-14 18:28:07 fv-az196-931 ocean_lib.data_provider.data_service_provider[9022] INFO Service initialized successfully initializeEndpoint http://172.15.0.4:8030/api/services/initialize
2022-10-14 18:28:12 fv-az196-931 ocean_lib.data_provider.data_service_provider[9022] INFO Service initialized successfully initializeEndpoint http://172.15.0.4:8030/api/services/initialize
2022-10-14 18:28:23 fv-az196-931 ocean_lib.data_provider.fileinfo_provider[9022] INFO Retrieved asset files successfully FileInfoEndpoint http://172.15.0.4:8030/api/services/fileinfo from did did:op:6a7c302f3de107d067b4282c106bd771baeb2710e32334d851258b37892dc144 with service id 0
------------------------------ Captured log call -------------------------------
INFO     ocean:ocean_assets.py:461 Successfully created NFT with address 0x99a64469d4bBaD1d79B3eb9BF131fAA8Ae25155a.
INFO     ocean:ocean_assets.py:546 Successfully created datatoken with address 0x9d9c6d87348c8345EF92AB77C9d30E7fB2A26c50.
INFO     ocean_lib.data_provider.data_encryptor:data_encryptor.py:58 Asset urls encrypted successfully, encrypted urls str: 0x04ec...3d766cff, encryptedEndpoint http://172.15.0.4:8030/api/services/encrypt
INFO     ocean_lib.data_provider.data_encryptor:data_encryptor.py:58 Asset urls encrypted successfully, encrypted urls str: 0x04cc...074f7615, encryptedEndpoint http://172.15.0.4:8030/api/services/encrypt
INFO     ocean_lib.data_provider.data_service_provider:data_service_provider.py:82 Service initialized successfully initializeEndpoint http://172.15.0.4:8030/api/services/initialize
INFO     ocean_lib.data_provider.data_service_provider:data_service_provider.py:82 Service initialized successfully initializeEndpoint http://172.15.0.4:8030/api/services/initialize
INFO     ocean_lib.data_provider.fileinfo_provider:fileinfo_provider.py:54 Retrieved asset files successfully FileInfoEndpoint http://172.15.0.4:8030/api/services/fileinfo from did did:op:6a7c302f3de107d067b4282c106bd771baeb2710e32334d851258b37892dc144 with service id 0
=============================== warnings summary ===============================
ocean_lib/web3_internal/test/test_contract_base.py::test_main
  /opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/site-packages/_pytest/threadexception.py:73: PytestUnhandledThreadExceptionWarning: Exception in thread Thread-1
  
  Traceback (most recent call last):
    File "/opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/threading.py", line 932, in _bootstrap_inner
      self.run()
    File "/opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/threading.py", line 870, in run
      self._target(*self._args, **self._kwargs)
    File "/opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/site-packages/enforce_typing/decorator.py", line 28, in wrapper
      check_types(*args, **kwargs)
    File "/opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/site-packages/enforce_typing/decorator.py", line 21, in check_types
      raise TypeError(
  TypeError: Expected type 'typing.Callable' for attribute 'callback' but received type '<class 'NoneType'>')
  
    warnings.warn(pytest.PytestUnhandledThreadExceptionWarning(msg))

tests/integration/remote/test_polygon.py::test_ocean_tx__create_url_asset
  /home/runner/work/ocean.py/ocean.py/tests/integration/remote/test_polygon.py:36: UserWarning: Warning: Insufficient Polygon MATIC
    warnings.warn(UserWarning("Warning: Insufficient Polygon MATIC"))

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED tests/integration/ganache/test_issue981_consume.py::test1 - ocean_lib....
= 1 failed, 234 passed, 6 skipped, 8 deselected, 2 warnings in 1500.77s (0:25:00) =
Error: Process completed with exit code 1.
```

**Comment by AnaLoznianu** [comment]
@calina-c Can we close this issue?

**Comment by calina-c** [comment]
Closed by #1075
